flake: TestDemoWorkerLifecycleScript_StartStop — worker subprocess exits early

## Symptom

`TestDemoWorkerLifecycleScript_StartStop` in `test/unit/demo_worker_lifecycle_script_test.go:64` fails intermittently on GitHub Actions with:

\`\`\`
--- FAIL: TestDemoWorkerLifecycleScript_StartStop (2.43s)
    demo_worker_lifecycle_script_test.go:64: start failed: exit status 1
        worker process exited early (see log: /tmp/TestDemoWorkerLifecycleScript_StartStop3210051656/004/demo-worker.log)
FAIL
FAIL  github.com/confighub/cub-scout/test/unit  12.169s
\`\`\`

## First observed

- **`v2.0.0` release workflow run** [24452684223](https://github.com/confighub/cub-scout/actions/runs/24452684223) — failed at the `Run tests` step after 1m36s, blocking the release pipeline. The same commit passed on the `v2.0.0-rc5` workflow run [24449063559](https://github.com/confighub/cub-scout/actions/runs/24449063559) 90 minutes earlier.
- Local `go test -run TestDemoWorkerLifecycleScript_StartStop -count=3 ./test/unit/` passes consistently in 0.57–0.90s per iteration.

Same commit, same test, different GitHub runner → pass. Classic flake.

## Analysis

The test spawns a child "demo worker" subprocess via a shell script, then polls for its PID or output. The "worker process exited early" branch fires when the child exits before the harness's first poll. Likely causes:

1. The worker's startup path depends on a file that is not yet written when the harness starts polling
2. The worker hits a transient failure (port in use, missing config, etc.) and exits cleanly
3. The harness polling window is too short for a cold GitHub runner

## Impact

This flake **blocks release promotion** when it hits on a `v*` tag push, because the `Run tests` step in `release.yaml` is a prerequisite for GoReleaser. Today (v2.0.0 release) it required a manual `gh run rerun --failed`.

## Proposed fix

Pick whichever root cause analysis confirms:

1. **Raise the startup grace period** before the harness starts polling (cheap, covers case 3)
2. **Read the worker log file** and assert on content rather than process state (covers case 2 — distinguishes "exited healthy early" from "exited with error")
3. **Write a temp lockfile when the worker is ready** and block the harness on it with a timeout
4. **Gate the test on `-tags=integration`** and skip it in the `release` workflow's `go test ./...` step — lowest-risk workaround if a proper fix is complex

## Workaround for the v2.0.0 release

Already applied: `gh run rerun 24452684223 --failed`. The rerun is expected to succeed because the test passes locally and passed on rc5's identical code.

Surfaced during v2.0.0 release promotion.

--- a/examples/scripts/demo-worker-lifecycle.sh
+++ b/examples/scripts/demo-worker-lifecycle.sh
@@ -234,7 +234,12 @@ start_worker() {
 
     echo "$worker_pid" >"$PID_FILE"
 
-    sleep 0.2
+    # Give the subprocess enough time to reach its own sleep/run loop before we
+    # check liveness.  0.2 s was too short on cold CI runners (GitHub Actions)
+    # where fork + exec latency can exceed the grace window, causing a false
+    # "worker process exited early" failure.  1 s is still fast enough for
+    # interactive demos while being robust on slow runners.  Tracked: #388.
+    sleep 1
     if ! kill -0 "$worker_pid" 2>/dev/null; then
         rm -f "$PID_FILE"
         echo "worker process exited early (see log: $LOG_FILE)" >&2

